src/artemis/parameters/plan_specific/grid_scan_with_edge_detect_params.py

```diff
@@ -34,14 +41,34 @@ def get_num_images(self):
 
 class GridScanWithEdgeDetectInternalParameters(InternalParameters):
     experiment_params: GridScanWithEdgeDetectParams
-    artemis_params: ArtemisParameters
+    artemis_params: GridscanArtemisParameters
 
     class Config:
         arbitrary_types_allowed = True
         json_encoders = {
             **ArtemisParameters.Config.json_encoders,
         }
 
+    def __init__(self, **args):
+        super().__init__(**args)
+
+    @staticmethod
+    def _artemis_param_key_definitions() -> tuple[list[str], list[str], list[str]]:
```

Comment:
Should: I feel like we've moved too much of this function to be in each different class, when only a small part of it is different for each. Could we only move the different part please?